src/DiagramGenerator/Image/StorageNew.php

```diff
@@ -0,0 +1,265 @@
+<?php
+
+namespace DiagramGenerator\Image;
+
+use DiagramGenerator\Config;
+use DiagramGenerator\Config\Texture;
+use DiagramGenerator\Fen;
+use DiagramGenerator\Fen\Piece;
+use Intervention\Image\Exception\NotReadableException;
+use Intervention\Image\Image;
+use Intervention\Image\ImageManagerStatic;
+use RuntimeException;
+
+class StorageNew implements StorageInterface
+{
+    protected $pieces = [];
+
+    /** @var string */
+    protected $cacheDirectory;
+
+    /**
+     * @param string $cacheDirectory
+     */
+    public function __construct($cacheDirectory)
```

Comment:
Please avoid docblock comments that only specify types. They are deprecated and redundant to strict types in the code.

```suggestion
    public function __construct(string $cacheDirectory)
```